modules/dnn/src/opencl/math.cl

```diff
@@ -0,0 +1,154 @@
+                                  __global Dtype* b,
+                                  const int offb, __global Dtype* y,
+                                  const int offy) {
+  for (int index = get_global_id(0); index < n; index += get_global_size(0)) {
+    y[index + offy] = a[index + offa] * b[index + offb];
+  }
+}
+
+__kernel void TEMPLATE(div,Dtype)(const int n, __global const Dtype* a,
+                                  const int offa,
+                                  __global Dtype* b,
+                                  const int offb, __global Dtype* y,
+                                  const int offy) {
+  for (int index = get_global_id(0); index < n; index += get_global_size(0)) {
+    y[index + offy] = a[index + offa] / b[index + offb];
+  }
+}
+
+__kernel void TEMPLATE(add_scalar,Dtype)(const int N, const Dtype alpha,
+__global Dtype* Y,
+                                         const int offY) {
+  for (int index = get_global_id(0); index < N; index += get_global_size(0)) {
+    Y[offY + index] += alpha;
+  }
+}
+
+__kernel void TEMPLATE(add,Dtype)(const int n, __global const Dtype* a,
+                                  const int offa, __global const Dtype* b,
+                                  const int offb, __global Dtype* y,
+                                  const int offy) {
+  for (int index = get_global_id(0); index < n; index += get_global_size(0)) {
+    y[offy + index] = a[offa + index] + b[offb + index];
+  }
+}
+
+__kernel void TEMPLATE(sub,Dtype)(const int n, __global const Dtype* a,
+                                  const int offa, __global const Dtype* b,
+                                  const int offb, __global Dtype* y,
+                                  const int offy) {
+  for (int index = get_global_id(0); index < n; index += get_global_size(0)) {
+    y[offy + index] = a[offa + index] - b[offb + index];
+  }
+}
+
+__kernel void TEMPLATE(abs,Dtype)(const int n, __global const Dtype* a,
+                                  const int offa, __global Dtype* y,
+                                  const int offy) {
+  for (int index = get_global_id(0); index < n; index += get_global_size(0)) {
+    y[offy + index] = fabs((Dtype)(a[offa + index]));
+  }
+}
+
+__kernel void TEMPLATE(exp,Dtype)(const int n, __global const Dtype* a,
+                                  const int offa, __global Dtype* y,
+                                  const int offy) {
+  for (int index = get_global_id(0); index < n; index += get_global_size(0)) {
+    y[offy + index] = exp(a[offa + index]);
+  }
+}
+
+__kernel void TEMPLATE(log,Dtype)(const int n, __global const Dtype* a,
+                                  const int offa, __global Dtype* y,
+                                  const int offy) {
+  for (int index = get_global_id(0); index < n; index += get_global_size(0)) {
+    y[offy + index] = log((Dtype)(a[offa + index]));
+  }
+}
+
+__kernel void TEMPLATE(powx,Dtype)(const int n, __global const Dtype* a,
```

Comment:
There is already OpenCV functions for functionality of these kernels. There is no strong reason to add duplicates.

Reply:
fixed in new updates. have removed these kernels and switch to opencv